pkg/build/generator/generator.go

```diff
@@ -702,3 +710,18 @@ func getStrategyImageChangeTrigger(bc *buildapi.BuildConfig) *buildapi.ImageChan
 	}
 	return nil
 }
+
+// getImageChangeTriggerForRef returns the ImageChangeTrigger that is triggered by a change to
+// the provided object reference, if any
+func getImageChangeTriggerForRef(bc *buildapi.BuildConfig, ref *kapi.ObjectReference) *buildapi.ImageChangeTrigger {
+	if ref == nil || ref.Kind != "ImageStreamTag" {
+		return nil
+	}
+	for _, trigger := range bc.Spec.Triggers {
+		if trigger.Type == buildapi.ImageChangeBuildTriggerType && trigger.ImageChange.From != nil &&
```

Comment:
no, that behavior isn't implemented...if an ICT has no from, it has to be referring to the strategy's from.

which kinda makes sense because otherwise it'd be impossible to know what a "default" ICT was supposed to point to (the strategy from, or one of the N imagesource Froms)